Signal/src/call/PeerConnectionClient.swift

```diff
@@ -128,17 +142,23 @@ class PeerConnectionClient: NSObject, RTCPeerConnectionDelegate, RTCDataChannelD
     // MARK: - Media Streams
 
     public func createSignalingDataChannel() {
-        let dataChannel = peerConnection.dataChannel(forLabel: Identifiers.dataChannelSignaling.rawValue,
-                                                     configuration: RTCDataChannelConfiguration())
-        dataChannel.delegate = self
+        AssertIsOnMainThread()
+        
+        PeerConnectionClient.signalingQueue.sync {
```

Comment:
This is the only exception: the main thread blocks on the `signalingQueue` when creating the data channel, to facilitate testing. Since `signalingQueue` never blocks on the main thread, there is no risk of dead lock.